cap-std/src/fs/dir.rs

```diff
@@ -609,6 +609,20 @@ impl Dir {
         let _ = ambient_authority;
         fs::create_dir_all(path)
     }
+
+    /// Construct a new instance of `Self` from existing directory file descriptor.
+    ///
+    /// This can be useful when interacting with other libraries and or C/C++ code
+    /// which has invoked `openat(..., O_DIRECTORY)` external to this crate.
+    #[cfg(not(windows))]
+    pub fn try_reopen_dirfd(fd: BorrowedFd) -> io::Result<Self> {
```

Comment:
Could we name this `reopen_dir`?
 - I think we can omit the `try_` because other "open" functions in cap-std and std don't have `try_` in their names, as it's commonly understood that opening something can fail.
 - Change "dirfd" to "dir" because I think we'll be able to generalize this to Windows in the future, with basically the same API, but without calling it "fd", so dropping the "fd" now will reduce churn later.

Reply:
Makes sense.  I think the reason I had `try_` in there is because I had in my head in the background that I also wanted to add a `try_exists` to cap-std (xref https://internals.rust-lang.org/t/the-api-of-path-exists-encourages-broken-code/13817)

And also related to that originally I was thinking of this as a "clone" operation, but it's a `try_clone()`.

But I agree `reopen_dir()` is clear and to the point, so fixed!